plugins/guests/redhat/cap/osvariant.rb

```diff
@@ -3,13 +3,12 @@ module GuestRedHat
     module Cap
       class OsVariant
         def self.os_variant(machine)
-          machine.communicate.sudo("grep VARIANT_ID /etc/os-release") do |type, data|
-            if type == :stderr
-              @env.ui.error(data)
-              exit 126
+          if machine.communicate.test("grep VARIANT_ID /etc/os-release")
+            machine.communicate.execute("grep VARIANT_ID /etc/os-release") do |type, data|
```

Comment:
This should do, however thinking out loud, for other boxes strings "CDK" or "ADB" could be a substring for other strings. Also this method does not check for ADB or CDK this just returns the VARIANT_ID value and check is happening somewhere else.